pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -292,17 +298,21 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
 
     LOGGER.debug("Broker Response: {}", brokerResponse);
 
-    // Table name might have been changed (with suffix _OFFLINE/_REALTIME appended)
-    LOGGER.info(
-        "RequestId:{}, table:{}, timeMs:{}, docs:{}/{}, entries:{}/{}, segments(queried/processed/matched):{}/{}/{} "
-            + "servers:{}/{}, groupLimitReached:{}, exceptions:{}, serverStats:{}, query:{}", requestId,
-        brokerRequest.getQuerySource().getTableName(), totalTimeMs, brokerResponse.getNumDocsScanned(),
-        brokerResponse.getTotalDocs(), brokerResponse.getNumEntriesScannedInFilter(),
-        brokerResponse.getNumEntriesScannedPostFilter(), brokerResponse.getNumSegmentsQueried(),
-        brokerResponse.getNumSegmentsProcessed(), brokerResponse.getNumSegmentsMatched(),
-        brokerResponse.getNumServersResponded(), brokerResponse.getNumServersQueried(),
-        brokerResponse.isNumGroupsLimitReached(), brokerResponse.getExceptionsSize(), serverStats.getServerStats(),
-        StringUtils.substring(query, 0, _queryLogLength));
+    // Sampling the query log based on the sampling rate configuration
+    if(_queryLogSamplingRate > RANDOM.nextFloat()) {
```

Comment:
maybe have other criteria which will force a log:
* if number of exceptions > 0
* timeMs > 100 (say)
* entries scanned post filter is high
* num groups limit is reached

I suggest adding a method shouldLog()

Reply:
@mcvsubbu Some numbers would be use case specific. For now, I have put some lenient numbers. We can tune this as we needed.